controller/run.go

```diff
@@ -74,3 +94,70 @@ func RunManaged(cfg config.ShipperConfig) error {
 
 	return err
 }
+
+// runController is the main runloop for the shipper itself, and managed communication with the agent.
+func runController(ctx context.Context, agentClient client.V2) error {
+	log := logp.L()
+
+	err := agentClient.Start(ctx)
+	if err != nil {
+		return fmt.Errorf("error starting connection to client")
+	}
+
+	log.Debugf("Starting error reporter")
+	go reportErrors(ctx, agentClient)
+
+	log.Debugf("Started client, waiting")
+	handler := newClientHandler()
+
+	sigc := make(chan os.Signal, 1)
+	signal.Notify(sigc, syscall.SIGINT, syscall.SIGTERM, syscall.SIGHUP)
+
+	// use a timer to "bounce" the shipper based on external changes
+	// useful since we require two units for the shipper to actually start, which can arrive at different times, or get updated with different states
+	// This also makes it a little easier to deal with multple events requiring
+	// a (re)start of the shipper backend, in which case we'll only restart once with the latest config.
+	bounceTime := 200 * time.Millisecond // give us enough time to get multiple units before the shipper takes any action
+	shipperSetTimer := time.NewTimer(bounceTime)
+	shipperSetTimer.Stop() // start with the timer off, reset when we get a change
+
+	// receive the units
+	for {
+		select {
+		case gotSignal := <-sigc:
+			switch gotSignal {
+			case syscall.SIGINT, syscall.SIGTERM:
+				handler.log.Info("Received sigterm/sigint, stopping")
+			case syscall.SIGHUP:
+				handler.log.Info("Received sighup, stopping")
+			}
+			handler.units.DeleteOutput()
+			shipperSetTimer.Reset(bounceTime)
+			return nil
+		case <-ctx.Done():
+			handler.log.Info("Got context done")
+			handler.units.DeleteOutput()
+			shipperSetTimer.Reset(bounceTime)
+			return nil
+		case change := <-agentClient.UnitChanges():
+			// Unsure of if we want to make some of these unit change operations parallel.
+			// Because we have to juggle state between two units, there's a lot of back-and-forth logic that can lead to races. Ditto with BounceShipper().
+			// for now, block while unit changes propigate instead of adding more mutexes.
+			restart := false
+			switch change.Type {
+			case client.UnitChangedAdded: // The agent is starting the shipper, or we added a new processor
```

Comment:
```suggestion
			case client.UnitChangedAdded: // The agent is starting the shipper
```

Let's not assume how the processor implementation will work until we actually do it, this seems like a way to end up with stale comments.